rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,611 @@
+- For web, we expose `duration` as a `string` of *milliseconds* which works well with CSS's `transition-duration` property but will not work with Framer motion since it requires a `number` in *seconds*
+- For web, we expose `easing` as a `string` of `cubic-bezier(..)` which works well with CSS's `transition-timing-function` property but will not work with Framer motion since it requires either,
+  - The name of an existing easing function.
+  - An array of four numbers to define a cubic bezier curve.
+  - An easing function, that accepts and returns a value 0-1.
+- For native, we expose `easing` as a function of `Easing.bezier(..)` imported from `react-native-reanimated` which won't work when used with React Native's `Animated` API.
+
+### React Native `v0.62`
+React Native Reanimated is dependent on `TurboModules`, it restricts us to using React Native `v0.62+` that supports `TurboModules`.
+
+### Multi-step keyframe animation with React Native
+Multi-step animations on web are implemented by having more than 2 steps in an animation. Read more about [multi-step animations here](https://www.joshwcomeau.com/animation/keyframe-animations/#multi-step-animations)
+
+Eg) 2-step keyframe
+```css
+@keyframes move {
+    0% {
+      transform: translateY(0px);
+    }
+    100% {
+      transform: translateY(100px);
+    }
+}
+```
+Eg) Multi-step keyframe
+```css
+@keyframes move {
+    0% {
+      transform: translateY(0px);
+    }
+    50% {
+      transform: translateY(50px);
+    }
+    100% {
+      transform: translateY(100px);
+    }
+}
+```
+
+- In the 2-step keyframe, easing is applied for the animation taking place from 0% to 100%.
+- In the multi-step keyframe, easing is applied for each step of the animation. That is, if we use `ease-in-out` easing, it would be applied to the animation happening from `0%` to `50%` and then again from `50%` to `100%`.
+- This is not supported in React Native Reanimated _directly_ since we apply easing to the `timing` function.
+
+  ```jsx
+    shakeAnimation.value = withTiming(100, { 
+      duration: motionToken.duration.duration3, 
+      easing: motionToken.easing.standard.effective, 
+    });
+  ```
+
+- We apply easing to the timing function as `shakeAnimation` value goes from 0 to 100. In this case if we apply an `ease-in-out` easing, it would be applied to the entire animation happening from `0%` to `100%` regardless of the fact that our interpolate's input & output range is more than 2 steps (array of more than 2 elements).
+- If we _really_ want to replicate the web's behaviour on react native, we would need to apply easing on the `interpolate` function. 
+- As of `v2.3.0` of `react-native-reanimated`, there is no support to add an easing to `interpolate`. 
+- To be able to achieve this, we would need to implement the animation using React native's `Animated` API since it supports applying easing to interpolation like this:
```

Comment:
so finally what are we concluding? we'll use Reanimated or native animated? above you have mentioned that we'll use reanimated but that has constraints of being used on `version: >0.62` and it also doesn't support the similar behavior to the animations on web.

Reply:
If we for some reason want to replicate the keyframes behaviour on react native, we'll use the Animated API. For all other use cases, react native reanimated works well for us. 

The constraint of the `0.62` version is not a big issue. All our internal apps above `0.62` and react native's latest is `0.68`